app/assets/javascripts/comfortable_mexican_sofa/admin/modules/LinkManager.js

```diff
@@ -40,10 +42,43 @@ define([
     this._initialisedSuccess(initialised);
   };
 
+  LinkManagerProto._setup = function(type, val) {
+    LinkManager.superclass._setup.call(this);
+    this._populateAnchors(val);
+  };
+
   LinkManagerProto._setupAppEvents = function() {
     eventsWithPromises.subscribe('cmseditor:insert-published', filterEvent($.proxy(this._handleShown, this), this.context));
     LinkManager.superclass._setupAppEvents.call(this);
   };
 
+  LinkManagerProto._populateAnchors = function(selectedValue) {
+    var self = this,
+        id,
+        text;
+
+    this._resetSelectOptions(this.config.selectors.anchors);
+
+    $(this.config.selectors.editorContents).find('h2, h3, h4').each(function() {
+      text = $(this).html();
+      id = '#' + self._slugify(text);
+      self._addSelectOption(self.config.selectors.anchors, id, text, selectedValue);
```

Comment:
Might want to re-jig this so that `_addSelectOptions` returns an HTML string (rather than doing the `.append`) that you build up in this method, and then write to the DOM only once.